### PR TITLE
lib/vfscore: Fix `getdents` count unit

### LINKED ISSUE #497: binary-compat: `opendir()` fails under binary compatibility mode
### Describe the bug

When running a static PIE executable under binary compatibility mode, using [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader), it errs when using `opendir()`:

```
opendir: Not a directory
```

### Steps to reproduce

Use [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader) to run a simple static PIE binary that uses `opendir()` such as [this one](https://github.com/unikraft/static-pie-apps/pull/11). Use the command below, assuming the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader) and the [`static-pie-apps` repository](https://github.com/unikraft/static-pie-apps/) are cloned in the same directory:

```
$ ./run_elfloader -r ../static-pie-apps/small/dir/rootfs/  ../static-pie-apps/small/dir/dir
```

The output shows the Unikraft boot message together with the error message on `opendir()`:

```
[...]
Powered by
o.   .o       _ _               __ _
Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
oOo oOO| | | | |   (| | | (_) |  _) :_
 OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
            Mimas 0.7.0~1e17a9c8-custom
opendir: Not a directory
```

To build your own setup, including Unikraft and aim to fix this bug, see [these instructions](https://unikraft.org/community/hackathons/2022-05-lyon/bincompat/#00-setup).

### Expected behavior

The simple program should start under binary compatibility mode and list folder contents in the root directory of the filesystem (passed using the `-r ...` option to [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader).

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by dinhngtu**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### Additional configuration

N/A

### Description of changes

libc expects the `count` parameter of `getdents` to represent the output buffer size.  Unikraft instead uses this as the number of directory entries to fill, significantly overfilling the output buffer.  This commit fixes `getdents` to use the correct interpretation of `count`.

A simple way to reproduce this problem is to list a directory with a few dozen files, preferably several times.

### CODE REVIEW

**Review by razvand — COMMENTED** [review]
Thanks, @dinhngtu . I've been meaning to get this PR at some point, I stumbled into the same problem when using [binary compatibility mode](https://github.com/unikraft/static-pie-apps/blob/master/small/dir/dir.c).

I'll wait for a review from @StefanJum and @RaduNichita and then I'll approve it.

**Review by razvand — COMMENTED** [review]
Awesome. Thanks, @dinhngtu . This latest version is OK. I had done the same thing [here](https://github.com/razvand/unikraft-bincompat/commit/df00ebd95077aa7aa783dba1db3ccd0fe7988d3c).

Now the update is compatible.

**Comment by dinhngtu** [comment]
Apologies, the patch still had a bug where the buffer would still get overfilled if `count` was not divisible by `sizeof(struct dirent)`.

**Review by StefanJum — APPROVED** [review]
Works great, thank you @dinhngtu.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Comment by craciunoiuc** [comment]
Why do I get notifications here even though I unsubscribed 2 times

😱 😱 😱

**Review by razvand — APPROVED** [review]
Thanks, @dinhngtu 

Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Comment by RaduNichita** [comment]
A bit late, but also works for me.

Reviewed-by: Radu Nichita [radunichita99@gmail.com](mailto:radunichita99@gmail.com)

**Pull request closed without merging**
